sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -225,16 +225,16 @@ class DeviceBinaryImage {
   public:
     using ValTy = std::remove_pointer<pi_device_binary_property>::type;
 
-    class ConstIterator
-        : public std::iterator<std::input_iterator_tag, // iterator_category
-                               ValTy,                   // value_type
-                               ptrdiff_t,               // difference_type
-                               const pi_device_binary_property, // pointer
-                               pi_device_binary_property>       // reference
-    {
+   class ConstIterator {
       pi_device_binary_property Cur;
 
     public:
+      using iterator_category = std::input_iterator_tag;
+      using value_type = ValTy;
+      using difference_type = ptrdiff_t;
+      using pointer = const pi_device_binary_property;
```

Comment:
I know this is the same as the original code, but... is it expected that this is not a pointer type? (And the reference is not a reference type?)

Reply:
In the pi.h is definition of struct _pi_device_binary_property_struct and there is the string:
https://github.com/intel/llvm/blob/5eaa14f803590ee900f06801c3e2669ee6b6c158/sycl/include/CL/sycl/detail/pi.h#L622
This is always the pointer type. Did I understand the question correctly?

Reply:
Thank you for pointing out the `typedef` -- I had missed that `pi_device_binary_property` was already a pointer type.